librz/analysis/arch/arm/arm_il32.c

```diff
@@ -252,6 +358,87 @@ static RzILOpBitVector *arg_mem(RzILOpBitVector *base_plus_disp, cs_arm_op *op,
 	return base_plus_disp;
 }
 
+/**
+ * Replicate given value to `dreg_width` length
+ * Note the ownership of `val` will be transfered
+ */
+static RzILOpBitVector *replicated_val(ut32 val_width, ut32 dreg_width, RzILOpBitVector *val) {
+	ut32 repeat_times = dreg_width / val_width;
+	if (dreg_width % val_width != 0) {
+		return NULL;
```

Comment:
`rz_warn_if_reached()`?

Reply:
You forgot this one @Heersin

Reply:
ok. btw rebase and squash some commits ?

Reply:
@Heersin yes, rebasing and squashing